fastly.go

```diff
@@ -34,6 +34,34 @@ func queryLoop(ctx context.Context, token, service string, m *prometheusMetrics,
 				contextSleep(ctx, time.Second)
 				continue
 			}
+			data := make(map[string]interface{})
+			body, _ := ioutil.ReadAll(resp.Body)
+			err = json.Unmarshal(body, &data)
+			if err != nil {
+			    return err
+			}
+			name, ok := data["name"].(string);
+			if !ok {
+				level.Error(logger).Log("err", "Received non-string type for service name")
+				contextSleep(ctx, time.Second)
+				continue
+			}
```

Comment:
We shouldn't lookup service name in the hot loop. This should be done once, at program start, and passed to queryLoop.

Reply:
I second this note. The service lookup in the hot loop is likely abusing the Fastly API, especially when running this with many (100+) services. (They've reached out to us about this issue in particular)